src/tests/perf.py

```diff
@@ -29,6 +29,8 @@ def __init__(self, config):
         self.ramp = config['test_config']['general']['fio_ramptime']
         self.duration = config['test_config']['general']['fio_runtime']
         self.min_iops = config['test_config']['perf_rand_read']['iops']
+        local_duration = config['test_config']['perf_rand_read'].get('runtime')
+        self.duration = (local_duration, self.duration)[local_duration is None]
```

Comment:
This is difficult to read.  I think you're just trying to use a default if the local_duration is not set.

In which case I'd probably just change this to:
```
self.duration = config['test_config']['perf_rand_read'].get('runtime', config['test_config']['general']['fio_runtime'])
```